subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/TaskSuccessfulMessage.java

```diff
@@ -21,4 +22,13 @@ public void execute(Task task) {
             }
         });
     }
+
+    public static void logOnSuccess(Task task, final Supplier<String> supplier) {
```

Comment:
Can you add short Javadoc why we need this method (and also in the other method on this class)?

Good stuff. I'm happy that you have figured out how to defer logging the message, until we have all information needed.